app/controllers/donations_controller.rb

```diff
@@ -1,32 +1,48 @@
 class DonationsController < ApplicationController
 
   def create
-    if session[:donations]
-      session[:donations] = session[:donations].merge(cart.add(params)) do |key, old_val, new_val|
-        (old_val.to_i + new_val.to_i).to_s
-      end
+    if validate_params
+      session[:donations] = validate_session
+      flash[:notice] = "Donation added to cart."
     else
-      session[:donations] = cart.add(params)
+      flash[:errors] = "Donation cannot be empty."
     end
-    flash[:notice] = "Donation added to cart."
     return_to_origin_page
   end
 
   private
 
-    def return_to_origin_page
-      if candidate_id
-        redirect_to(candidate_path(candidate_id))
-      else
-        redirect_to(issue_path(issue_id))
-      end
+  def return_to_origin_page
+    if candidate_id
+      redirect_to(candidate_path(candidate_id))
+    else
+      redirect_to(issue_path(issue_id))
     end
+  end
 
-    def candidate_id
-      params[:donation][:candidate_id]
-    end
+  def candidate_id
+    params[:donation][:candidate_id]
+  end
+
+  def issue_id
+    params[:donation][:issue_id]
+  end
 
-    def issue_id
-      params[:donation][:issue_id]
+  def create_cart
+    cart.create(params)
+  end
+
+  def validate_params
+    params[:donation][:amount].to_i > 0
+  end
+
+  def validate_session
+    if session[:donations]
+      session[:donations].merge(create_cart) do |key, old_val, new_val|
```

Comment:
Unused block argument - `key`. If it's necessary, use `_` or `_key` as an argument name to indicate that it won't be used.